app/javascript/sprocket_asset_import/editor/editor.js

```diff
@@ -389,7 +391,7 @@ var CodeOceanEditor = {
     updateEditorModeToFileTypeID: function (editor, fileTypeID) {
         var newMode = 'ace/mode/text'
 
-        $.ajax(this.fileTypeURL + '/' + fileTypeID, {
+        $.ajax(this.fileTypeURL() + '/' + fileTypeID, {
```

Comment:
Routes.js needs the page to be rendered before execution. Calling this on execution ensures the page is present.